### PR TITLE
[camera] Fix camera memory leak on IOS

### PR DESCRIPTION
**Pull request by greggor01**
Needed to unregister the stream handler when disposing of the camera

## Description

Per the flutter documentation, the stream handler should be unregistered using setStreamHandler with nil.
https://api.flutter.dev/objcdoc/Classes/FlutterEventChannel.html#/c:objc(cs)FlutterEventChannel(im)setStreamHandler:

## Related Issues
Possibly fixes:
flutter/flutter#29586

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by greggor01** [comment]
Any news on when this might be incorporated?  We have an app that will be posted in the app store very soon and it would be nice to have this fix included since it does indeed cause a crash on some IOS devices if the camera is opened and closed multiple times.

File: packages/camera/ios/Classes/CameraPlugin.m [hunk 1]
```diff
@@ -732,6 +732,8 @@ - (void)handleMethodCallAsync:(FlutterMethodCall *)call result:(FlutterResult)re
     } else if ([@"dispose" isEqualToString:call.method]) {
       [_registry unregisterTexture:textureId];
       [_camera close];
+      // need to unregister stream handler when disposing the camera
+      [_camera.eventChannel setStreamHandler:nil];
```
[the remark this answers is not in the document]
Comment (greggor01) [reply packages/camera/ios/Classes/CameraPlugin.m]: @mklim I think you are correct that the onCancel handler is the correct place to release the stream handler instead on dispose.  I have made that change.

File: packages/camera/pubspec.yaml [hunk 2]
```diff
@@ -2,8 +2,9 @@ name: camera
 description: A Flutter plugin for getting information about and controlling the
   camera on Android and iOS. Supports previewing the camera feed, capturing images, capturing video,
   and streaming image buffers to dart.
-version: 0.5.2+1
-publish_to: none
+
```
Comment (mklim) [inline packages/camera/pubspec.yaml]: nit: this empty line isn't needed

**Comment by JoseGeorges8** [comment]
Just tested the camera plugin version 0.5.2+2 on an iphone 5S running OS 12.3 and after every time the camera opens and closes the memory usage keeps on increasing quickly (took about 10 times for the camera to open and close to terminate the app)

**Pull request merged**